fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java

```diff
@@ -201,6 +201,32 @@
      */
     void setInitialDelayMillis(long initialDelayMillis);
 
+    /**
+     * Set the transaction timeout for all transactions started by this runner. If set to {@link FDBDatabaseFactory#DEFAULT_TR_TIMEOUT_MILLIS},
+     * then this will use the value of set in the originating database's factory. If set to {@link FDBDatabaseFactory#UNLIMITED_TR_TIMEOUT_MILLIS},
+     * then no timeout will be imposed on transactions used by this runner.
+     *
+     * <p>
+     * Note that the error that the transaction hits, {@link FDBExceptions.FDBStoreTransactionTimeoutException},
+     * is not retriable, so if the runner encounters such an error, it will terminate.
+     * </p>
+     *
+     * @param transactionTimeoutMillis the transaction timeout time in milliseconds
+     * @see FDBDatabaseFactory#setTransactionTimeoutMillis(long)
+     */
+    void setTransactionTimeoutMillis(long transactionTimeoutMillis);
+
+    /**
+     * Get the transaction timeout for all transactions started by this runner. This will return the value configured
+     * for this runner through {@link #setTransactionTimeoutMillis(long)}. Note, however, that if the transaction timeout
+     * is set to {@link FDBDatabaseFactory#DEFAULT_TR_TIMEOUT_MILLIS}, then the actual timeout set for this transaction
+     * will be set to the value in the originating factory.
+     *
+     * @return the configured transacation timeout time in milliseconds
+     * @see #setTransactionTimeoutMillis(long)
+     */
+    long getTransactionTimeoutMillis();
```

Comment:
These are "breaking changes" for implementors of the interface, if we care about that use case. They are fine for those that are using the interface.

Reply:
Maybe the release notes should call out this minor (unlikely) breaking change (I did something like this on 2.8.103.0:

* **Breaking change** `ByteScanLimiter` and `RecordScanLimiter` are now interfaces. Instances with various concrete behavior are constructed through factory classes. [(PR #836)](https://github.com/FoundationDB/fdb-record-layer/pull/836)

Reply:
Okay, added release notes calling it out. For these two methods, I attached the issue (#848) as they seemed "inherent" to that Issue. I also call out the slight change in semantics regarding transaction IDs, but with this PR, as it seemed more "incidental".